### DISCUSSION TITLE
how to generate pre-deployed contract in genesis.json?

### DISCUSSION THREAD

**Discussion by JSHan94** [0xPolygon/polygon-edge#424] [Q&A] [answered]
Hello,

We are trying to deploy a contract in pre-defined address with Polygon-edge.

After some digging, we found that we can set our contract in genesis.json file.

Here is what we set,

![스크린샷 2022-02-18 오후 12 59 42](https://user-images.githubusercontent.com/29887117/154615172-e7535940-3894-49f4-8181-bca3f38aef72.png)

But, when we interact this contract with Web3.js, it doesn't work well like below as follow.

![스크린샷 2022-02-18 오후 1 44 27](https://user-images.githubusercontent.com/29887117/154619120-b1872117-d89c-4579-9f57-63b9e87b8988.png)

We compiled our contract with Remix and set the bytecode to genesis.json.

we also set storage data.

Is there any tutorial for deploying contract in specific address?

**Comment by Kourin1996** [comment, marked as answer]
@JSHan94 Hello, Thank you for your questions. If you set pre-deploy contract in genesis.json, you need to set `deployedBytecode` in `code` field of pre-deploy account setting in `genesis.json`. Would you mind telling me which bytecode you set? Thank you.

**Reply by JSHan94** [reply]
We want to set pre-deploy contract on ```0x00.....0010001``` address like the above figure. We add ```deployedBytecode``` to ```code``` field. we set the ```storage``` field as we saw during compiling contract and debugging process at Remix.
Am i doing right? It's hard to find pre-deploy contract tutorials..

**Reply by Kourin1996** [reply]
It's possible to pre-deploy contracts by setting the code and storage data. But we don't have the explanation in our document, unfortunately.

As your screenshot said, the error occurred because revert was called in contract while transaction execution. There are possible various reasons. The contract may have reverted against invalid arguments (normal behavior) or failed loading storage (wrong initial storage). It's difficult to find correct reason without knowledge of how the smart contract works, what you wanted to set to initial storage.
